pkg/controllers/updaterun/execution.go

```diff
@@ -55,8 +59,9 @@ func (r *Reconciler) execute(
 	updatingStageIndex int,
 	toBeUpdatedBindings, toBeDeletedBindings []*placementv1beta1.ClusterResourceBinding,
 ) (bool, time.Duration, error) {
-	// Mark the update run as started regardless of whether it's already marked.
-	markUpdateRunStarted(updateRun)
+	// Before the execute returns, mark the update run as started in memory if it's not marked as waiting or stuck.
+	// If the update run actually succeeds or fails, it will be remarked later in controller.go.
+	defer markUpdateRunStartedIfNotWaitingOrStuck(updateRun)
```

Comment:
using "defer" statement looks weird to me. Marking updateRun as started is more natural at the beginning if it's not marked as waiting or stuck.

If error happens, the condition will be reset.

Reply:
The idea is, during the exection, we may mark it as stuck or waiting. After one round of execution, if it's not marked as stuck or waiting, that means the updateRun is still progressing and this call ensures that.